test/xpu_api/utilities/meta/meta.trans/meta.trans.other/has_type_member.h

```diff
@@ -0,0 +1,32 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _HAS_TYPE_MEMBER_H
+#define _HAS_TYPE_MEMBER_H
+
+#include <oneapi/dpl/type_traits>
+
+template <class, class = dpl::void_t<>>
+struct has_type_member : dpl::false_type
+{
+};
+
+template <class T>
+struct has_type_member<T, dpl::void_t<typename dpl::underlying_type<T>::type>> : dpl::true_type
```

Comment:
I guess it should be simply `T` here. Otherwise, `has_type_member` does not work exactly how it was for `result_of.pass.cpp`.

Reply:
```suggestion
struct has_type_member<T, dpl::void_t<typename T::type>> : dpl::true_type
```

Reply:
Fixed. But I have checked both tests and they compiled OK.